ingest/src/main/java/com/microsoft/azure/kusto/ingest/ExponentialRetry.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.azure.kusto.ingest;
+
+public class ExponentialRetry {
```

Comment:
nit: I think they're both valid design decisions so I'm not requesting a change here, but I personally would've passed to this class a `Function` that makes the call + error handling logic, and `doBackoff()` actually kicks off making the call repeatedly with backoff.
I just see that approach as better encapsulated.

Reply:
It would be kind of annoying here because in the body of the loop we do a `return`, so we'll need to make it work with generics and everything.
I think the current approach is good enough for now.